Automattic/wp-calypso
# Components: Use compact form toggle instead of the default large one

As per p9Jlb4-1Q9-p2 we're working to unify our toggles and get them to use `@wordpress/components`. The first step is to get rid of the large toggle, which happens to be the default in Calypso. This PR updates all of the instances of the default large toggle to use a compact one.

We're consciously ignoring any further improvements or moving of code in order to make the PR easier to review.

Furthermore, we're fixing some ESLint rules when passing by.

#### Changes proposed in this Pull Request

* Use compact toggle instead of the general toggle component.

#### Testing instructions

Go to each of these, and follow the instructions. Verify we're now using a compact toggle and it looks and works well. Each step has screenshots where applicable.

* `/settings/taxonomies/category/:site` - click "Add New Category" and observe the "Top level Category" toggle. 
Before:
	![](https://cldup.com/ONHCvL5GL1.png)
After:
	![](https://cldup.com/kbzI-RDC1U.png)
* `/devdocs/design/form-fields` - the three Form Toggle examples that now represent all possible form toggle states (previously five when we had the large form toggle).
Before:
	![](https://cldup.com/NhpY_G6laM.png)
After:
	![](https://cldup.com/DvfWes4O7F.png)
* `/devdocs/design/g-suite` - the two form toggles at the bottom of the example. Note that we're also updating the labels to be used properly as form toggle labels as an additional improvement.
Before:
	![](https://cldup.com/h2irHxYana.png)
After:
	![](https://cldup.com/X4qET_BnL2.png)
* `/devdocs/playground` - try inserting a `<FormToggle />` in the DevDocs playground and you should see a working compact toggle.
* `/store/product/:siteSlug/:productId`, where `:siteSlug` is the site slug of a site with WooCommerce, and `:productId` is the ID of a product, observe the "product with variations..." toggle.
Before:
	![](https://cldup.com/cbz1PvaClh.png)
After:
	![](https://cldup.com/_TiGyH1gSk.png)
* `/store/product/:siteSlug/:productId`, where `:siteSlug` is the site slug of a site with WooCommerce, and `:productId` is the ID of a product with variations; click on the name of the variation and in the modal, observe the "Visible" toggle.
Before:
	![](https://cldup.com/25M8r3iL12.png)
After:
	![](https://cldup.com/B5wwOSiBK4.png)
* `/store/settings/taxes/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce and WooCommerce Services active, and a valid USA address selected as the store address, observe the "Tax calculations enabled" toggle.
Before:
	 ![](https://cldup.com/ZQztlz9TVF.png)
After:
	 ![](https://cldup.com/gM6vAV7WEr.png)
* `/store/settings/shipping/:siteSlug` where where `:siteSlug` is the site slug of a site with WooCommerce and WooCommerce Services active, observe the toggle at the right of the "Shipping Labels" card heading.
Before:
	 ![](https://cldup.com/HeATbLzQEf.png)
After:
	 ![](https://cldup.com/z4PuSJBV89.png)
* `/me/privacy` - the toggle at the bottom of the "Usage Information" card.
Before:
	![](https://cldup.com/nxgDwIcDmp.png)
After:
	![](https://cldup.com/RLA3c9jDio.png)
* `/me/purchases` - select a purchase, and observe the toggle in the Subscription Renewal section (the rest of the auto renew toggle instances in the domain management section were already compact ones).
Before:
	![](https://cldup.com/xTvMl1w3cb.png)
After:
	![](https://cldup.com/F3rIJxfRnj.png)
* `/domains/manage/:siteSlug/contacts-privacy/:domain` where `:siteSlug` your site and `:domain` is a domain on that site, observe the "Privacy Protection" toggle
Before:
	![](https://cldup.com/JXkBoIWFAY.png)
After:
	![](https://cldup.com/XHX1xAZEQd.png)
* `/domains/manage/:siteSlug/dns/:domain` where `:siteSlug` is your site and `:domain` is a domain on that site, observe the domain connect toggle. It's possible it can be disabled if the domain is managed by WP.com currently:
Before:
	![](https://cldup.com/oPGogqq1y1.png)
After:
	![](https://cldup.com/buTMsgNnYT.png)
* `/domains/manage/:siteSlug/name-servers/:domain` where `:siteSlug` your site and `:domain` is a domain on that site, observe the "Use WordPress.com Name Servers" toggle
Before:
	![](https://cldup.com/b8m_XUcpCo.png)
After:
	![](https://cldup.com/lTe8O6v0Qo.png)
* `/earn/payments-plans/:site` where `:site` is the slug of a simple WP.com site with a business plan. You will have to authorize Stripe first before accessing that screen. When there, click on the "Add a new payment plan" button, and observe the two toggles in the modal:
Before:
	![](https://cldup.com/GFCNrUZhe2.png)
After:
	![](https://cldup.com/2QrPj_u75H.png)
